### PR TITLE
feat: thorchain evm transaction parsing

### PR DESCRIPTION
**Pull request by kaladinlight**
## Description

Adds thorchain transaction parsing for all supported evm chains

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes https://github.com/shapeshift/web/issues/6020

Low - isolated to evm tx parsers. make sure evm transactions are being displayed correctly

## Testing

Ensure the transaction row is displaying the correct type and information for the following thorchain types on evm chains (both native asset and tokens):
- Swap Request (Standard and Streaming)
- Swap Received (Standard and Streaming)
- Swap Refund (Standard and Streaming)
- Liquidity Deposit (Savers and LP when we can test)
- Liquidity Refund (Savers and LP when we can test)
- Liquidity Withdraw Request (Savers and LP when we can test)
- Liquidity Withdraw Received (Savers and LP when we can test)
- Loan Open
- Loan Open Refund
- Loan Repayment
- Loan Repayment Refund
- Loan Collateral Received
- Swap Request Longtail (no associated tag unless we would like that @reallybeard)

Also worth spot checking evm transaction history to validate everything else is still being displayed as expected.

### Engineering

:point_up:

### Operations

:point_up:

## Screenshots (if applicable)

**Swaps**
image
image
image
image
image

**Streaming Swaps**
image
image

image
image
image

**Savers**
image
image
image
image

**LP**
image
image
image

**Loan**
image
image
image
image

**Longtail**
image
image

### CODE REVIEW

File: packages/unchained-client/src/evm/parser/thorchain.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+const routerAbi = [
+  'function deposit(address vault, address asset, uint256 amount, string memo)',
+  'function depositWithExpiry(address vault, address asset, uint256 amount, string memo, uint256 expiration)',
+  'function transferOut(address to, address asset, uint256 amount, string memo)',
+  'function transferOutAndCall(address target, address finalAsset, address to, uint256 amountOutMin, string memo)',
+  'function swapIn(address tcRouter, address tcVault, string tcMemo, address token, uint256 amount, uint256 amountOutMin, uint256 deadline)',
+]
+
+const depositRegex = /^(add|a|\+):([^:]+)(:([^:]+)?)?(:([^:]+)?)?(:([^:]+)?)?$/
+const withdrawRegex = /^(withdraw|wd|-):([^:]+)(:([^:]+)?)?(:([^:]+)?)?$/
```
Comment (gomesalexandre) [inline packages/unchained-client/src/evm/parser/thorchain.ts:18]: So simple 🐐

File: packages/unchained-client/src/evm/parser/thorchain.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+      if (Object.values(this.supportedFunctions).some(hash => hash === txSigHash)) {
+        const decoded = this.abiInterface.parseTransaction({ data: tx.inputData })
+
+        // failed to decode input data
+        if (!decoded) return
+
+        return (decoded.args.memo as string) || (decoded.args.tcMemo as string)
+      }
+
+      // input data may be a raw thorchain memo
```
Comment (gomesalexandre) [inline packages/unchained-client/src/evm/parser/thorchain.ts:69]: 💜

File: packages/unchained-client/src/parser/thorchain.ts [hunk 3]
```diff
@@ -0,0 +1,251 @@
+import type { Axios } from 'axios'
+import axios from 'axios'
+
+import type { BaseTxMetadata, StandardTx } from '../types'
+import { Dex, TradeType } from '../types'
+
+export type LiquidityType = 'Savers' | 'LP'
```
Comment (gomesalexandre) [inline packages/unchained-client/src/parser/thorchain.ts:7]: 👀

**Review by gomesalexandre — APPROVED** [review]
Tested locally with a subset of the added parsing which I have available, because of testing funds - deferring to @shapeshift/operations for the green light on this

 ### LP Deposit
 
image

### LP Withdraw requested

image

### LP Withdraw Received 

image

### Swap

image

### Savers withdraw request and withdraw received

- Note that withdraw request isn't parsed obviously since it's a send vs. contract call

image

 ### Savers EVM token deposit
 
image

### Loan Collateral Received

image
image

### Loan Open

image

### Long-tail

image
image

**Comment by MBMaria** [comment]
Non Blocker: 

- [ ] Still getting a couple transactions listed as "Transaction" in the Activity tab

image
image

**Comment by MBMaria** [comment]
Testing thread: https://discord.com/channels/554694662431178782/1201332574547607592

**Pull request merged**